src/gd_filter.c

```diff
@@ -15,7 +15,12 @@
 #include <time.h>
 #include <math.h>
 
-#define NDEBUG  /* Uncomment to enable assertions. */
+#ifndef NDEBUG
```

Comment:
this change doesn't make sense.  the spec for assert.h specifically says that it's allowed to be included multiple times and NDEBUG can change value every time.  so you've effectively disabled the logic we had here to disable assertions.  it also means the double assert.h inclusion is now pointless.  see the documentation:
http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/assert.h.html

if you're trying to avoid redefinitions of NDEBUG, then add an #undef line before the #define.  that'll force consistent behavior no matter how NDEBUG is set.